go/cmd/dolt/commands/checkout.go

```diff
@@ -56,59 +56,81 @@ func Checkout(ctx context.Context, commandStr string, args []string, dEnv *env.D
 	if (apr.Contains(coBranchArg) && apr.NArg() > 1) || (!apr.Contains(coBranchArg) && apr.NArg() == 0) {
 		usagePrt()
 		return 1
+	}
+
+	var verr errhand.VerboseError
+	if apr.ContainsArg(doltdb.DocTableName) {
+		verr = errhand.BuildDError("Use dolt checkout to checkout individual docs.").Build()
+		cli.PrintErrln(verr.Verbose())
+		return 1
+	}
+
+	if newBranch, newBranchOk := apr.GetValue(coBranchArg); newBranchOk {
+		return checkoutNewBranch(ctx, dEnv, newBranch, apr)
+	}
+
+	name := apr.Arg(0)
+
+	if isBranch, err := actions.IsBranch(ctx, dEnv, name); err != nil {
+		verr = errhand.BuildDError("error: unable to determine type of checkout").AddCause(err).Build()
+		cli.PrintErrln(verr.Verbose())
+		return 1
+	} else if isBranch {
+		return checkoutKnownBranch(ctx, dEnv, name)
+	}
+
+	tbls, docs, err := actions.GetTblsAndDocDetails(dEnv, args)
+	if err != nil {
+		verr = errhand.BuildDError("error: unable to parse arguments.").AddCause(err).Build()
+		cli.PrintErrln(verr.Verbose())
+		return 1
+	}
+
+	verr = checkoutTablesAndDocs(ctx, dEnv, tbls, docs)
+
+	if verr != nil && apr.NArg() == 1 {
+		verr = tryCheckoutNewBranch(ctx, dEnv, name)
+	}
+
+	if verr != nil {
+		cli.PrintErrln(verr.Verbose())
+		return 1
+	}
+
+	return 0
+}
+
+func tryCheckoutNewBranch(ctx context.Context, dEnv *env.DoltEnv, name string) errhand.VerboseError {
```

Comment:
Would call this checkoutRemoteBranch

Reply:
Renamed: https://github.com/liquidata-inc/dolt/pull/299/files#diff-78081aa9ea2b0558c0f56aad2a236961R95